### PR TITLE
Only log but don't exit on error

### PR DESCRIPTION
**Pull request by MrHohn**
Ref https://github.com/kubernetes-incubator/ip-masq-agent/pull/22#issuecomment-427141115. It makes more sense to log the error and retry, instead of crashing the container and let kubelet handle the restart:
- Restart doesn't help resolving most of the situations (e.g. iptables lock is hold).
- Unnecessary crashlooping generates false alerts given this is likely a kube-system pod.

Fixes https://github.com/kubernetes-incubator/ip-masq-agent/issues/23.

/assign @jingax10

cc @negz

### CODE REVIEW

**Comment by negz** [comment]
LGTM, thank you!

File: cmd/ip-masq-agent/ip-masq-agent.go [hunk 1]
```diff
@@ -121,31 +121,21 @@ func main() {
 }
 
 func (m *MasqDaemon) Run() error {
```
[the remark this answers is not in the document]
Comment (MrHohn) [reply cmd/ip-masq-agent/ip-masq-agent.go]: Good catch! Fixed now.

**Comment by MrHohn** [comment]
Merging this per LGTM on https://github.com/kubernetes-incubator/ip-masq-agent/pull/24#issuecomment-427152087

**Pull request merged**